plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/search/SearchResultIterator.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.datafusion.search;
+
+import java.util.Iterator;
+
+/**
+ * Interface for the search results iterator
+ */
+public interface SearchResultIterator extends Iterator<Record> {
+    // Basic Iterator methods
+    boolean hasNext();
+
+    Record next();
+}
```

Comment:
The interface looks subset of `RecordBatchStream`, except that iterates on `RecordBatch` instead of `Record`? Probably, we should add more comprehensive documentation for each interface and its purpose for easier consumption and understanding